integrated_channels/cornerstone/utils.py

```diff
@@ -31,3 +50,26 @@ def create_cornerstone_learner_data(request, course_id):
         pass
     except Exception as ex:  # pylint: disable=broad-except
         LOGGER.error('Unable to Create/Update CornerstoneLearnerDataTransmissionAudit. {ex}'.format(ex=ex))
+
+
+def convert_invalid_course_ids(course_id):
+    """
+    Regex check a course ID to see if it contains any invalid chars. If it does then encode the string, otherwise
+    return the original course ID.
+    """
+    re2 = re.compile(r"[|<>.&%\s\\/\“]+")
+    if re2.search(course_id):
+        # If the course key contains any of the invalid chars, encode the key
+        course_id = encode_course_key_into_base64(course_id)
+    # If the encoded or unencoded version of the key are over 50 characters, they will error out
+    # in cornerstone, so we convert them to a uuid.
+    if len(course_id) > 50:
+        course_id = uuid4()
+    return course_id
+
+
+def add_cornerstone_key_pair(course_id):
+    key_mapping, ___ = cornerstone_course_key().objects.get_or_create(
+        internal_course_id=course_id, defaults={
```

Comment:
formatting?

Reply:
It's weird but it's the way lint wanted 🙄